src/concurrency/vector_clock.rs

```diff
@@ -77,16 +79,15 @@ impl VTimestamp {
     }
 
     fn set_time(&mut self, time: u32) {
-        self.time_and_read_type =
-            (self.time_and_read_type & 1) | time.checked_mul(2).expect("Vector clock overflow")
+        self.time_and_read_type = Self::new(time, self.read_type(), DUMMY_SP).time_and_read_type;
```

Comment:
Please define an `encode_time_and_read_type` function, that seems better than creating a full `VTimestamp` and then only reading one field and avoids the dummy span.